photon-client/src/main/scala/com/linkedin/photon/ml/cli/game/scoring/Params.scala

```diff
@@ -242,6 +251,16 @@ object Params {
             "convention.")
         .foreach(x => params.offHeapIndexMapNumPartitions = x)
 
+      opt[String]("input-columns-names")
+        .text(s"A map of the input columns names, in format: response:label|offset:offset|uid:uid ... ")
+        .foreach(x =>
+          x.split("\\|")
+            .foreach { line =>
+              val Array(key, value) = line.split(":").map(_.trim)
+              params.inputColumnsNames.updated(InputColumnsNames.withName(key), value)
+            }
+        )
+
```

Comment:
I prefer having individual parameters because there's a small number of them, and our algorithms depend on them directly. I.e. we'd never add or remove one without some change to the code. So a dynamic structure seems out of place to me.

The way they handle this in spark.ml is to have traits for each column that mix the setters/getters into the Estimator. E.g.:

```scala
estimator
  .setLabelCol("response")
  .setWeightCol("weight")
  .setOffsetCol("offset")
```